### ISSUE TITLE
Elim of V types?

### ISSUE THREAD

**Issue by favonia** [RedPRL/sml-redprl#475]

**Comment by favonia** [comment]
@cangiuli What should be the left rule of V types? I am not sure about the first component. We should be able to say, with the restriction `r = 0`, there should be an element in the domain `A`, but this formulation (with restrictions) does not seem to work directly in a sequent calculus.

**Comment by cangiuli** [comment]
I feel that we should never expose context restrictions to the user, because nobody should be able to or have to prove goals under an assumption that `r = 0`. Instead, I would like to treat every `r` as a parameter to the rule. Further subgoals should not be generated until a concrete `r` is provided. Is there any problem with this?

This is also related to #184. There I believe that we should be able to see subgoals only when `r`, `r'`, and all tube equations are provided.

**Comment by favonia** [comment]
If it is a right rule, then it is easy. The problem is for left rules in a sequent calculus, you need to fill in the `blah` part
```
G, a:V(...), blah, D[Vin(...)/a] >> J[Vin(...)/a]
--------------------------------------------------
G, a:V(...), D >> J
```
and this seems very difficult.

**Comment by cangiuli** [comment]
I want to have three separate rules, for `0`, `1`, and `x`. Is it still difficult?

**Comment by favonia** [comment]
Yes---how would you handle the variable case? The well-formedness of `blah` only holds under `G[0/x]`, not `G`. (For the `1` case, `b` can just be garbage.)

**Comment by favonia** [comment]
By the way, I am leaning toward handling only the variable cases even in `V.EqProj`. For non-variable cases the type synthesizer might not give us something in the form `V(...)`, and it is probably too troublesome to work around this limitation.

**Comment by favonia** [comment]
The title was updated because `V.EqProj` was just implemented by #607.
